metrics/cloudwatch/cloudwatch.go

```diff
@@ -44,6 +46,15 @@ func New(namespace string, svc cloudwatchiface.CloudWatchAPI, logger log.Logger)
 	}
 }
 
+// SetConcurrency overrides the default number (10) of concurrent requests sent to CloudWatch.
+// CloudWatch allows a maximum of 20 metrics to be sent per request, so when Send is called,
+// we partition the metrics and concurrently call their API. This parameter sets maximum number
+// of concurrent requests.
+func (cw *CloudWatch) SetConcurrency(numConcurrentRequests int) *CloudWatch {
+	cw.numConcurrentRequests = numConcurrentRequests
+	return cw
+}
```

Comment:
This should be an option in the constructor, rather than a method on the struct. I don't see any reason to make it modifiable during runtime. (It's also racy as implemented.)